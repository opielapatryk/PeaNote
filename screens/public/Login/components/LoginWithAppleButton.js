import React from 'react';
import {Pressable, Text} from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';
import { styles } from '../../../../assets/styles/styles';
import {signIn} from '../functions/signIn'
import * as AppleAuthentication from 'expo-apple-authentication';

const LoginWithAppleButton = () => {
    return (
        <AppleAuthentication.AppleAuthenticationButton
        buttonType={AppleAuthentication.AppleAuthenticationButtonType.SIGN_IN}
        buttonStyle={AppleAuthentication.AppleAuthenticationButtonStyle.BLACK}
        cornerRadius={5}
        style={styles.continuteWithGoogle}
        onPress={async () => {
          try {
            const credential = await AppleAuthentication.signInAsync({
              requestedScopes: [
                AppleAuthentication.AppleAuthenticationScope.FULL_NAME,
                AppleAuthentication.AppleAuthenticationScope.EMAIL,
              ],
            });
            // signed in
          } catch (e) {
            if (e.code === 'ERR_REQUEST_CANCELED') {
              // handle that the user canceled the sign-in flow
            } else {
              // handle other errors
            }
          }
        }}
      />
    );
}

export default LoginWithAppleButton;