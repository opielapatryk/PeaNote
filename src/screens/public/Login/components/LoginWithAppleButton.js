import React from 'react';
import { styles } from '../../../../assets/styles/styles';
import * as AppleAuthentication from 'expo-apple-authentication';
import {loginWithAppleFunction} from '../functions/loginWithAppleFunction'

const LoginWithAppleButton = () => {
    return (
        <AppleAuthentication.AppleAuthenticationButton
        buttonType={AppleAuthentication.AppleAuthenticationButtonType.SIGN_IN}
        buttonStyle={AppleAuthentication.AppleAuthenticationButtonStyle.BLACK}
        cornerRadius={5}
        style={styles.continuteWithGoogle}
        onPress={loginWithAppleFunction}
      />
    );
}

export default LoginWithAppleButton;