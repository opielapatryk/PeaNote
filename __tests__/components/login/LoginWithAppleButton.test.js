import React from 'react';
import renderer from 'react-test-renderer';
import LoginWithAppleButton from '../../../src/screens/login/components/LoginWithAppleButton';

test('LoginWithApple button renders correctly.', () => {
  const tree = renderer.create(<LoginWithAppleButton />).toJSON();
  expect(tree).toMatchSnapshot();
});