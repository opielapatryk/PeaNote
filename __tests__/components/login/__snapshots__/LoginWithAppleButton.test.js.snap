// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithApple button renders correctly. 1`] = `
<ViewManagerAdapter_ExpoAppleAuthentication
  buttonStyle={2}
  buttonType={0}
  cornerRadius={5}
  onButtonPress={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFDF3",
      "borderColor": "lightgray",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 1,
      "flexDirection": "row",
      "height": 66.7,
      "justifyContent": "center",
      "paddingLeft": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
      "width": 500,
    }
  }
/>
`;
